yt/utilities/answer_testing/framework.py

```diff
@@ -63,6 +81,332 @@
+        return result
+
+    def small_patch_amr(self, ds, field, weight, axis, ds_obj):
+        hex_digests = {} 
+        # Grid hierarchy test
+        gh_hd = self.grid_hierarchy_test(ds)
+        hex_digests['grid_hierarchy'] = gh_hd
+        # Parentage relationships test
+        pr_hd = self.parentage_relationships_test(ds)
+        hex_digests['parentage_relationships'] = pr_hd
+        # Grid values, projection values, and field values tests
+        gv_hd = self.grid_values_test(ds, field)
+        hex_digests['grid_values'] = gv_hd
+        fv_hd = self.field_values_test(ds, field, ds_obj)
+        hex_digests['field_values'] = fv_hd
+        pv_hd = self.projection_values_test(ds, axis, field, weight, ds_obj)
+        hex_digests['projection_values'] = pv_hd
+        return hex_digests
+
+    def big_patch_amr(self, ds, field, weight, axis, ds_obj):
+        hex_digests = {} 
+        # Grid hierarchy test
+        gh_hd = self.grid_hierarchy_test(ds)
+        hex_digests['grid_hierarchy'] = gh_hd
+        # Parentage relationships test
+        pr_hd = self.parentage_relationships_test(ds)
+        hex_digests['parentage_relationships'] = pr_hd
+        # Grid values, projection values, and field values tests
+        gv_hd = self.grid_values_test(ds, field)
+        hex_digests['grid_values'] = gv_hd
+        ppv_hd = self.pixelized_projection_values_test(ds, axis, field, weight, ds_obj)
+        hex_digests['pixelized_projection_values'] = ppv_hd 
+        return hex_digests
+
+    def generic_array_test(self, func, args=None, kwargs=None):
+        if args is None:
+            args = []
+        if kwargs is None:
+            kwargs = {}
+        return func(*args, **kwargs)
+
+    def sph_answer(self, ds, ds_str_repr, ds_nparticles, field, weight, ds_obj, axis):
+        # Make sure we're dealing with the right dataset
+        assert str(ds) == ds_str_repr
+        # Set up keys of test names
+        hex_digests = {} 
+        dd = ds.all_data()
+        assert dd["particle_position"].shape == (ds_nparticles, 3)
+        tot = sum(dd[ptype, "particle_position"].shape[0]
+                  for ptype in ds.particle_types if ptype != "all")
+        # Check
+        assert tot == ds_nparticles
+        dobj = utils.create_obj(ds, ds_obj)
+        s1 = dobj["ones"].sum()
+        s2 = sum(mask.sum() for block, mask in dobj.blocks)
+        assert s1 == s2
+        if field[0] in ds.particle_types:
+            particle_type = True
+        else:
+            particle_type = False
+        if particle_type is False:
+            ppv_hd = self.pixelized_projection_values_test(ds, axis, field, weight, ds_obj)
+            hex_digests['pixelized_projection_values'] = ppv_hd
+        fv_hd = self.field_values_test(ds, field, ds_obj, particle_type=particle_type)
+        hex_digests['field_values'] = fv_hd
+        return hex_digests
+
+    def yt_field_test(self, ds, field, geometric):
```

Comment:
maybe call this `test_field_size_and_mean`? `yt_field_test` indicates to me that this is doing something with all of the field values in the test. Plus we have plenty of other field tests in here.